### PR TITLE
Chapter Dependency Injection

### LINKED ISSUE #31: [anheumai-dparys] Dependency Injection
Complete the Dependency Injection lesson

### PR DESCRIPTION
**Pull request by dariuszparys**
Closes #31

### CODE REVIEW

File: anheumai-dparys/di/di.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+func Greet(w io.Writer, name string) {
+	fmt.Fprintf(w, "Hello, %s", name)
+}
```
Comment (oliverscheer) [inline anheumai-dparys/di/di.go:11]: nit: My linter is complaining about missing comments ;-)

File: anheumai-dparys/di/di_test.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	"testing"
+)
+
+func TestGreet(t *testing.T) {
+	buffer := bytes.Buffer{}
+	Greet(&buffer, "Dariusz")
+
+	got := buffer.String()
+	want := "Hello, Dariusz"
+
```
Comment (oliverscheer) [inline anheumai-dparys/di/di_test.go:14]: I like your way of personalizing code.

**Review by oliverscheer — APPROVED** [review]

File: anheumai-dparys/di/di.go (hunk 1 above)
Comment (dariuszparys) [reply anheumai-dparys/di/di.go:11]: yes, I had comments on the first couple of lines in the first chapter but I became so sarcastic that I decided to leave them out, I'm currently reading how to configure linting, I don't like this senseless
_Do a comment and it has to start with the name of the commenting function, type or whatever_

Comment (oliverscheer) [reply anheumai-dparys/di/di.go:11]: Agree. I comment sometimes with // NameOfFunction Lorem Ipsum ... :-)

**Pull request closed without merging**